src/main/java/lol/server/Game.java

```diff
@@ -0,0 +1,101 @@
+package lol.server;
+
+import java.net.*;
+import java.io.*;
+import java.util.*;
+import lol.common.*;
+import lol.game.*;
+import lol.game.action.*;
+import lol.ui.*;
+
+public class Game {
+
+  private ArrayList<Player> players;
+  private Arena arena;
+  private Battlefield battlefield;
+  private LOL2D ui;
+
+  public Game(LOL2D ui, Player bluePlayer, Player redPlayer, Battlefield battlefield) {
+    this.ui = ui;
+    players = new ArrayList<>();
+    players.add(bluePlayer);
+    players.add(redPlayer);
+    this.battlefield = battlefield;
+    arena = new Arena(battlefield);
+  }
+
+  public void startGame() throws IOException {
+    championSelectionPhase();
+    spawnPhase();
+    arena.startGamePhase();
+    ui.update();
+    wait(2000);
+  }
+
+  public void gameLoop() throws IOException {
+    for(int turns = 0; battlefield.allNexusAlive() && turns < lol.client.Client.MAX_TURNS; ++turns) {
+      for(Player player : players) {
+        Turn turn = player.askTurn();
+        arena.applyTurn(turn);
+        ui.update();
+        broadcast(turn);
+        wait(2000);
```

Comment:
I'm not sure we want to wait that long if it's a tourney, depending on the numbers of AI how much time a single tourney can take?